### PR TITLE
Add functions to retrieve streams info and format in json 

### PR DESCRIPTION
**Pull request by xadhoom**

### CODE REVIEW

**Comment by talklittle** [comment]
Thanks for the PR. I understand the point of this change is to get values parsed as numbers etc. instead of all being binaries. It's probably worth making it a breaking change to the existing `format/1` function instead of making a separate function. After all, I can't think of a use case for keeping the binary values. It's only that I wasn't aware ffprobe could output JSON.

File: lib/ffprobe.ex [hunk 1]
```diff
@@ -52,6 +53,26 @@ defmodule FFprobe do
...
+  def format_json(file_path) do
+    cmd_args = ["-v", "quiet", "-print_format", "json", "-show_format", file_path]
+    {result, 0} = System.cmd ffprobe_path(), cmd_args, stderr_to_stdout: true
+
+    result
+    |> Poison.decode!()
+    |> Map.get("format", %{})
+  end
+
+  @spec streams_json(binary) :: format_json
```
Comment (talklittle) [inline lib/ffprobe.ex]: Please rename `streams_json` -> `streams` and make the typespec clear that it returns a list.

**Comment by xadhoom** [comment]
Thanks for yout feedbakc!

I've (forced) pushed the PR, now also format/1 uses json format from ffprobe, and fixed the typespecs.

This introduce a breaking change in the format/1 api, maybe is worth annotate it somewhere.

File: lib/ffprobe.ex [hunk 2]
```diff
@@ -41,15 +42,24 @@ defmodule FFprobe do
   Get the "format" map, containing general info for the specified file,
   such as number of streams, duration, file size, and more.
   """
-  @spec format(binary) :: format_map
+  @spec format(binary) :: format_map | no_return
```
Comment (talklittle) [inline lib/ffprobe.ex]: These always return a value, even if it might be an empty map/list, so please remove `no_return`

File: lib/ffprobe.ex [hunk 3]
```diff
@@ -9,7 +9,8 @@ defmodule FFprobe do
   (from https://trac.ffmpeg.org/wiki/FFprobeTips)
   """
 
-  @type format_map :: %{binary => binary}
+  @type format_map :: Poison.Poison.Parser.t
+  @type streams_list :: [Poison.Poison.Parser.t]
```
Comment (talklittle) [inline lib/ffprobe.ex]: `Poison.Poison.Parser.t` -> `Poison.Parser.t`

File: test/ffprobe_test.exs [hunk 4]
```diff
@@ -16,11 +16,19 @@ defmodule FFprobeTest do
...
-    assert %{"nb_streams" => "2"} = FFprobe.format(@fixture)
+    assert %{"nb_streams" => 2} = FFprobe.format(@fixture)
   end
 
   test "format which filename has space" do
-    assert %{"nb_streams" => "2"} = FFprobe.format(@fixture_space)
+    assert %{"nb_streams" => 2} = FFprobe.format(@fixture_space)
+  end
+
+  test "streams_json/1" do
```
Comment (talklittle) [inline test/ffprobe_test.exs]: Please rename test to reflect updated function name `streams/1`

**Comment by xadhoom** [comment]
Pushed (force) like requested, except for the `no_return` which is correct.

File: lib/ffprobe.ex (hunk 2 above)
Comment (talklittle) [reply lib/ffprobe.ex]: You're right, thanks.

**Comment by xadhoom** [comment]
Thanks!

**Pull request merged**